packages/x-charts/src/ChartsText/ChartsText.tsx

```diff
@@ -24,14 +25,17 @@ function ChartsText(props: ChartsTextProps) {
 
   const { angle, textAnchor, dominantBaseline, ...style } = styleProps ?? {};
 
+  const isClient = useIsClient();
+
   const wordsByLines = React.useMemo(
-    () => getWordsByLines({ style, needsComputation: text.includes('\n'), text }),
-    [style, text],
+    () => getWordsByLines({ style, needsComputation: isClient && text.includes('\n'), text }),
+    [style, text, isClient],
   );
```

Comment:
I noticed similar hydration issues when using `\n` in axes labels. SO I added this fix that ignore the size computation during the hydration phase

Reply:
By the way, technically `isClient` is not correct, because the hydration is on the client. It's more `isHydrationDone`

Reply:
> By the way, technically isClient is not correct, because the hydration is on the client. It's more isHydrationDone

I'll rename it in a follow-up.

Reply:
Done: https://github.com/mui/mui-x/pull/16937